curriculum/challenges/english/25-front-end-development/lab-real-time-counter/66bb3e20d3dc5b6d0a21f5dd.md

```diff
@@ -0,0 +1,202 @@
+2. There should be a `p` element with the `id` of `char-count`. It should initially contain the text `Character Count: 0/50`. This placeholder text should be replaced with the current count of characters in the `textarea` element.
+3. When the `#text-input` element contains the text `hello coder` the `#char-count` element should contain the text `"Character Count: 11/50"`
+4. When the `#text-input` element contains the text `i am a programmer` the `#char-count` element should contain the text `"Character Count: 17/50"`
+5. When the `#text-input` element contains the text `hello world` the `#char-count` element should contain the text `"Character Count: 11/50"`
+6. When the `#text-input` element contains the text `I am learning a new language and it is called c++.` the `#char-count` element should contain the text `"Character Count: 50/50"`.
+7. The user should not be able to enter more than `50` characters in the `textarea` element. When the character count reaches `50`, the text `Character Count: 50/50` should be displayed in `red`.
+
+**Note:** Be sure to link your stylesheet and the JavaScript file in your HTML.
+
+# --hints--
+
+You should have a `textarea` element with the `id` of `text-input`. 
+
+```js
+const textInput = document.querySelector('textarea');
+assert.equal(textInput.id, 'text-input');
+```
+
+You should have a `p` element with the `id` of `char-count`. 
+
+```js
+const charCount = document.querySelector('p');
+assert.equal(charCount.id, 'char-count');
+```
+
+The `#char-count` `p` element should initially contain the text `Character Count: 0/50`.
+
+```js
+assert(document.querySelector('p')?.innerText, 'Character Count: 0/50');
+```
+
+When the `#text-input` element contains the text `hello coder` the `#char-count` element should contain the text `"Character Count: 11/50"`
+
+```js
+const textInput = document.getElementById('text-input');
+const charCount = document.getElementById('char-count');
+textInput.value = "hello coder";
+textInput.dispatchEvent(new Event('input'));
+textInput.dispatchEvent(new Event('change'));
+assert.strictEqual(charCount.innerText.trim(), 'Character Count: 11/50');
+```
+
+When the character count is `50`, the text should be displayed in `red`.
+
+```js
+const textInput = document.getElementById('text-input');
+const charCount = document.getElementById('char-count');
+textInput.value = "I am learning a new language and it is called c++.";
+textInput.dispatchEvent(new Event('input'));
+textInput.dispatchEvent(new Event('change'));
+assert.strictEqual(charCount.style.color, 'red');
+```
+
+If character count is greater than `50`, the user shouldn't be able to enter more characters.
+
+```js
+let textInput = document.getElementById("text-input");
+let charCount = textInput.value.length;
+textInput.value = "I am learning a new language and it is called c++.";
+textInput.dispatchEvent(new Event('input'));
+textInput.dispatchEvent(new Event('change'));
+assert.strictEqual(charCount, 50);
```

Comment:
```suggestion
textInput.value = "I am learning a new programming language and it is called c++.";
textInput.dispatchEvent(new Event('input'));
textInput.dispatchEvent(new Event('change'));
assert.strictEqual(textInput.value.length, 50);
```

This test wasn't working quite right. The string you had was exactly 50 characters, we want to make sure that a string longer than 50 isn't allowed.